### PR TITLE
Astroquery module for ISO archive

### PR DESCRIPTION
**Pull request by jesusjuansalgado**
Added new astroquery submodule called:
astroquery.esa.iso

This module allows access to the ESA mission ISO (Infrared Space Observatory) data:

https://www.esa.int/Science_Exploration/Space_Science/ISO_overview

ISO (1995-2001) is an important ESA legacy mission that observed the sky in the
infrared wavelength. These data have special interest these days as it is being used
by scientists e.g. to help on the preparation JWST proposals, both because of the
energy range that ISO covered and for transient data identification.

### CODE REVIEW

File: astroquery/esa/iso/__init__.py [hunk 1]
```diff
@@ -25,8 +25,8 @@ class Conf(_config.ConfigNamespace):
 
     TIMEOUT = 60
 
-conf = Conf()
```
Comment (jesusjuansalgado) [inline astroquery/esa/iso/__init__.py]: Code format

**Review by jesusjuansalgado — COMMENTED** [review]
Increase coverage

**Comment by jesusjuansalgado** [comment]
Added new astroquery submodule called:
astroquery.esa.iso

File: astroquery/esa/iso/core.py [hunk 2]
```diff
@@ -109,6 +145,48 @@ def download_data(self, tdt, *, retrieval_type=None, filename=None,
 
         return filename
 
+    def get_postcard_link(self, tdt, filename=None, verbose=False):
```
Comment (bsipocz) [inline astroquery/esa/iso/core.py]: filename is not used, and docstring need a cleanup of irrelevant parameter names

**Comment by jesusjuansalgado** [comment]
All checks passed

File: astroquery/esa/iso/core.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+        job = self._tap.launch_job(query=query, output_file=output_file,
+                                   output_format=output_format,
+                                   verbose=verbose,
+                                   dump_to_file=output_file is not None)
+        try:
+            table = job.get_results()
+            return table
+        except HTTPError:
+            print(table)
```
Comment (keflavich) [inline astroquery/esa/iso/core.py:265]: this block is not valid: if `job.get_results()` raises an `HTTPError`, `table` will not be defined.

**Review by keflavich — CHANGES_REQUESTED** [review]
I have a few minor change requests, mostly related to style.

The changes to astropy-helpers, hubble, and xmm shouldn't be included in this PR, though.

File: CHANGES.rst [hunk 4]
```diff
@@ -10,6 +10,12 @@ mast
 - Added ``Observations.download_file`` method to download a single file from MAST given an input
   data URI. [#1825]
 
+esa/iso
+  ^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (jesusjuansalgado) [reply CHANGES.rst:14]: Corrected

File: astroquery/esa/iso/core.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    def __init__(self, tap_handler=None):
+        super(ISOClass, self).__init__()
+
+        if tap_handler is None:
+            self._tap = TapPlus(url=self.metadata_url)
+        else:
+            self._tap = tap_handler
+
+    def get_download_link(self, tdt, retrieval_type, filename,
+                      verbose, **kwargs):
```
[the remark this answers is not in the document]
Comment (jesusjuansalgado) [reply astroquery/esa/iso/core.py:45]: Corrected and also reviewed other similar problems

File: astroquery/esa/iso/core.py [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+        _, params = cgi.parse_header(response.headers['Content-Disposition'])
+        r_filename = params["filename"]
+        suffixes = Path(r_filename).suffixes
+
+        if filename is None:
+            filename = tdt
+
+        filename += "".join(suffixes)
+
+        log.info("Copying file to {0}...".format(filename))
```
[the remark this answers is not in the document]
Comment (jesusjuansalgado) [reply astroquery/esa/iso/core.py:139]: All log.info hidden by verbose

File: astroquery/esa/iso/core.py [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+        retrieval_type : string
+            type of retrieval: OBSERVATION for full observation or STANDALONE
+            for single files
+        filename : string
+            file name to be used to store the file
+        verbose : bool
+            optional, default 'False'
+            flag to display information about the process
+
+         Returns
```
[the remark this answers is not in the document]
Comment (jesusjuansalgado) [reply astroquery/esa/iso/core.py:172]: Corrected whitespace for Results on some methods

File: astroquery/esa/iso/core.py [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+
+        local_filepath = self._request('GET', link, cache=True, save=True)
+
+        if filename is None:
+            try:
+                response = self._request('HEAD', link)
+                response.raise_for_status()
+            except requests.exceptions.RequestException as e:
+                print(str(e))
+                raise SystemExit(e)
```
[the remark this answers is not in the document]
Comment (jesusjuansalgado) [reply astroquery/esa/iso/core.py:219]: Corrected as suggested

File: astroquery/esa/iso/core.py [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+            of whether it is simple or composite.
+        filename : string
+            file name to be used to store the postcard, optional, default None
+        verbose : bool
+            optional, default 'False'
+            Flag to display information about the process
+
+        Returns
+        -------
+        None. It downloads the observation postcard indicated
```
[the remark this answers is not in the document]
Comment (jesusjuansalgado) [reply astroquery/esa/iso/core.py:206]: Returns corrected as method output

File: astroquery/esa/iso/core.py (hunk 3 above)
Comment (jesusjuansalgado) [reply astroquery/esa/iso/core.py:265]: print exception instead of possible not defined table

**Comment by jesusjuansalgado** [comment]
> I have a few minor change requests, mostly related to style.
> 
> The changes to astropy-helpers, hubble, and xmm shouldn't be included in this PR, though.

astropy-helpers and hubble, xmm changes removed from PR

**Comment by jesusjuansalgado** [comment]
@keflavich I implemented there required changes a couple of weeks ago but I am not sure if I needed to notify in some way that the pull request is ready to be review again. Please let me know

**Comment by bsipocz** [comment]
@jesusjuansalgado - There are still unrelated changes included in the diff that needs to be removed, best done during a rebase as there are also some merge commits.
I can help with that during next week, when I'll be able to finish up the review I started for this.

**Comment by jesusjuansalgado** [comment]
> @jesusjuansalgado - There are still unrelated changes included in the diff that needs to be removed, best done during a rebase as there are also some merge commits.
> I can help with that during next week, when I'll be able to finish up the review I started for this.

Thanks Brigitta!

**Pull request closed without merging**